### DISCUSSION TITLE
Help us sponsor the libraries we use

### DISCUSSION THREAD

**Discussion by binford2k** [community/community#10139] [Other Feature Feedback, Questions, & Ideas]
The log4j thing has made it clear that depending on open source developers to align with our product goals is unsustainable and unfair. We would like to pay them for their efforts. But it's hard to have a complete picture of all the projects and libraries we use. Right now, we have to crawl through every repo and build a list and rank and sort all the dependencies.

But Dependabot already knows all our dependencies. Why can't it do the ranking and sorting for us and then surface it on a single org page that shows all the libraries we use, ranked by how many repos, how active the repos are, etc. And each library on the page should have a link to its sponsor page so that we can easily and effectively sponsor the developers who have the greatest impact at our company.

**Comment by binford2k** [comment]
I'm not sure how I missed it, but there's a good start at https://github.com/orgs/puppetlabs/insights/dependencies.

I'd like to see a few more filtering, sorting, grouping options, but it's an awesome start.

* should be able to filter out dependencies from our own org
* should be able to group dependencies by namespace, so if we have a ton of minor dependencies on small tools from one author then that can be surfaced in aggregate
* should be able to show only repos with sponsorship opportunities
* would be great to sort by impact, which could be defined as the activity of each repo affecting the rank of the dependency
* should have a way of specifying our "high value" repositories and then bubbling their dependencies to the top. That way we make sure that the dependencies that have the most commercial value to us are compensated appropriately and fairly for that.

Even better would be if it were able to recurse and identify transitive dependencies. That way, if the dependencies of our project all depended on some lower level library, the fact that we were, in aggregate, highly dependent on that library would be surfaced.

I think this page is a great start, but it appears to be mostly designed with mitigating security concerns in mind. I would love it if there were another view that was built with sponsorship in mind.

**Comment by raphink** [comment]
Note: Organization insights is currently only available to Enterprise Cloud organisations as [per docs](https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/viewing-insights-for-your-organization).
